packages/core/src/bodl/create-bodl-service.ts

```diff
@@ -0,0 +1,40 @@
+import BodlService from "./bodl-service";
+import NoopBodlService from "./noop-bodl-service";
+import BodlEmitterService from "./bodl-emitter-service";
+import { isBodlEnabled } from "./is-bodl-enabled";
+import { CheckoutService } from "../checkout";
+import { MissingDataError, MissingDataErrorType } from "../common/error/errors";
+
+/**
+ * Creates an instance of `BodlService`.
+ *
+ * @remarks
+ * 
+ * ```js
+ * const bodlService = BodlService();
+ * bodlService.checkoutBegin();
+ * 
+ * ```
+ *
+ * @param {CheckoutService} checkoutService - An instance of CheckoutService
+ * @returns an instance of `BodlService`.
+ */
+export default function createBodlService(
+    checkoutService: CheckoutService,
```

Comment:
Instead of checkoutService can we pass the store here?

Reply:
Which injects all this in the background https://github.com/bigcommerce/checkout-sdk-js/blob/master/packages/core/src/checkout/create-checkout-service.ts

Reply:
I'm going to use here in the same way https://github.com/bigcommerce/checkout-js/blob/42ee27d679f81eaabc8b43771b08575525962d4a/packages/core/src/app/checkout/CheckoutApp.tsx#L67
So, checkoutService will be available for me

Reply:
Ahh let me come back on this tomorrow and see if there is a better way to do this.

Reply:
Hey @jairo-bc , was able to check with David today and we came with a solution where should be able to pass the state https://github.com/animesh1987/checkout-sdk-js/commit/ba48907aa9448a56a6fa4e41b8915acb83dff32a. This should simplify a lot of code in bodl-emitter-service. 

Please feel free to reach out if you have any questions.

cc @davidchin @bc-peng